### ISSUE TITLE
meryl failed after 2 attempts

### ISSUE THREAD

**Issue by RxLoutre** [marbl/canu#187]
Hi ! I'm a new canu user, it's my first time using it.

I tried this command  : 
Loutre:~/canu-1.3/Linux-amd64/bin$ perl canu -assemble -p susuzkii_canu -d /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly genomeSize=150000000 -pacbio-raw /media/loutre/DATAPART1/Documents/suzukii_assembly/pacbio_reads/filtered_subreads.fastq

At first, it seemed to work, but then, at a point something goes wrong , here the error message I got after a while : 

canu failed with 'failed to generate mer counts.  Made 2 attempts, jobs still failed'.

Any ideas of why I could did wrong ? Thanks for your answers !

Cheers,

Roxane

**Comment by colindaven** [comment]
Hi,

I am not sure if you did the first two stages correct and trim first, see usage below. It would appear better to start with something like this:

~/canu-1.3/Linux-amd64/bin$ canu -p susuzkii_canu -d /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly genomeSize=150000000 -pacbio-raw

That might help. If not, search issues for meryl specific errors and check the meryl logs.

**Comment by RxLoutre** [comment]
So, the first two stages I should do first is trimming and an other step ?
I don't understand your answer, I saw that you removed the input file and the -assemble step.
Should I use first the -correct and then -trim step ? And then, give the output of theeses two steps to the -assemble step ?
Also where I can find the meryl logs ?

**Comment by skoren** [comment]
By default Canu consists of three stages, correct, trim, and assemble (see the quick start here:http://canu.readthedocs.io/en/stable/quick-start.html#assembling-pacbio-data). By explicitly specifying -assemble you are bypassing the first two and assembling the uncorrected PacBio reads directly. What @colindaven suggested is not specifying -assemble so the full pipeline runs as default.

**Comment by RxLoutre** [comment]
Okay, thanks you a lot for your quick answers ! That should work now, sorry for dumb question !

**Comment by RxLoutre** [comment]
Well, the correction step finished without any issues, but I still get the same error, 
here the meryl log : sh: 1: /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts/meryl.sh: Permission denied

It seems that I don't have the exec right on meryl.sh : 
-rw------- 1 loutre loutre  123 juil.  6 15:49 meryl.000001.out
-rw------- 1 loutre loutre 1423 juil.  6 15:49 meryl.sh
 And I can't modify it using chmod, even with being superuser.

Any ideas ?

**Comment by skoren** [comment]
Have you run any of the quick-start datasets to make sure Canu is installed and working on your system. When Canu creates meryl.sh it will run chmod to make it executable but if your chmod is failing when you run it it likely failed within Canu as well. 

What error does chmod report? What is the full log from Canu up to the crash?

**Comment by RxLoutre** [comment]
No I didn't try any quick-start datasets, I'm going to try the one you gived below.

Chmod did not even report any error when I manually tried it.

The full log in my terminal show me that the Correction step have passed well : 

BEGIN CORRECTION
----------------------------------------
 Starting command on Wed Jul  6 15:37:00 2016 with 147.2 GB free disk space

    /home/loutre/canu-1.3/Linux-amd64/bin/gatekeeperCreate \
      -minlength 1000 \
      -o /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/susuzkii_canu.gkpStore.BUILDING \
      /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/susuzkii_canu.gkpStore.gkp \
    > /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/susuzkii_canu.gkpStore.BUILDING.err 2>&1
 Finished on Wed Jul  6 15:48:12 2016 (672 seconds) with 136.4 GB free disk space

Then, it start analysing  reads (I can't show you the log because it will take to much space, but it show the read length histogram, with the stars).

And then it starts meryl as showed : 

 Meryl attempt 1 begins.
----------------------------------------
 Starting concurrent execution on Wed Jul  6 15:49:03 2016 with 136.3 GB free disk space (1 processes; 2 concurrently)

    /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts/meryl.sh 1 > /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts/meryl.000001.out 2>&1

Finished on Wed Jul  6 15:49:03 2016 (lickety-split) with 136.3 GB free disk space
----------------------------------------

 meryl failed.

 Meryl attempt 2 begins.
----------------------------------------
 Starting concurrent execution on Wed Jul  6 15:49:03 2016 with 136.3 GB free disk space (1 processes; 2 concurrently)

    /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts/meryl.sh 1 > /media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts/meryl.000001.out 2>&1

 Finished on Wed Jul  6 15:49:03 2016 (lickety-split) with 136.3 GB free disk space
----------------------------------------

 meryl failed.

And that's all I've got, maybe I'm missing some log files somewhere else that could help you ?

**Comment by RxLoutre** [comment]
One strange thing is that meryl is working on the ecoli dataset provided in the quickstart, and in like 13 seconds.

**Comment by skoren** [comment]
You said "It seems that I don't have the exec right on meryl.sh...And I can't modify it using chmod, even with being superuser."

Which is why I was asking about chmod, are you able to modify meryl.sh to be executable? If not, that is the source of your issue and needs to be fixed. I would also make sure that you remove the canu_assembly folder completely before trying to re-run.

**Comment by RxLoutre** [comment]
No I can't do it, here what I tried : 

Loutre:/media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts$ chmod u+x meryl.sh 
Loutre:/media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts$ ls -l
total 5
-rw------- 1 loutre loutre  123 juil.  6 15:49 meryl.000001.out
-rw------- 1 loutre loutre 1423 juil.  6 15:49 meryl.sh
Loutre:/media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts$ chmod 777 meryl.sh 
Loutre:/media/loutre/DATAPART1/Documents/suzukii_assembly/canu_assembly/correction/0-mercounts$ ls -l
total 5
-rw------- 1 loutre loutre  123 juil.  6 15:49 meryl.000001.out
-rw------- 1 loutre loutre 1423 juil.  6 15:49 meryl.sh

Okay, I tried to run again after deleting the whole canu_assembler folder. I'm waiting for the end of the analysis.

**Comment by RxLoutre** [comment]
So, I solved my issue.

The problem of rights was because I couldn't use chmod in the /media/loutre/DATAPART1 device (which is a partition of my hard drive).

Who knows why I can't modify execution right in there, and why chmod don't give any error message.

But the fact is that I solved it by changing the output canu_assembly directory in ~/Documents/canj_assembly instead.

Thanks for your answers ! Have a nice day !

Cheers,

Roxane
